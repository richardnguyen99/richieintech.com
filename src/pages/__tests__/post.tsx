/**
 * Testing suite for the Post page
 *
 * @author Richard Nguyen <richard.ng0616@gmail.com>
 */
import React from "react";
import { StaticQuery } from "gatsby";
import renderer from "react-test-renderer";

import { BlogQueryQuery } from "@generated/graphql";
import Post from "../post";

beforeEach(() => {
  (StaticQuery as jest.Mock<StaticQuery>).mockImplementation(({ render }) =>
    render({
      latest: {
        edges: {
          node: {
            fields: {
              slug: "/",
            },
            frontmatter: {
              title: "Default Title",
              tags: ["tags"],
              description: "Default description",
              categories: ["blogs"],
              date: "2020-09-06",
              thumbnail: {
                childImageSharp: {
                  fluid: {
                    aspectRatio: 2,
                    src: `test_image.jpg`,
                    srcSet: `some srcSet`,
                    srcSetWebp: `some srcSetWebp`,
                    sizes: `(max-width: 600px) 100vw, 600px`,
                    base64: `string_of_base64`,
                  },
                },
              },
            },
            excerpt: undefined,
            timeToRead: undefined,
          },
        },
      },
    })
  );
});

describe("Index page", () => {
  it("matches snapshot", () => {
    const data: BlogQueryQuery = {
      latest: {
        edges: [
          {
            node: {
              fields: {
                slug: "/",
              },
              frontmatter: {
                title: "Default Title",
                tags: ["tags"],
                description: "Default description",
                categories: ["blogs"],
                date: "2020-09-06",
                thumbnail: {
                  childImageSharp: {
                    fluid: {
                      aspectRatio: 2,
                      src: `test_image.jpg`,
                      srcSet: `some srcSet`,
                      srcSetWebp: `some srcSetWebp`,
                      sizes: `(max-width: 600px) 100vw, 600px`,
                      base64: `string_of_base64`,
                    },
                  },
                },
              },
              excerpt: `Some to read`,
              timeToRead: 1,
            },
          },
        ],
      },
    };
    const tree = renderer.create(<Post data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
