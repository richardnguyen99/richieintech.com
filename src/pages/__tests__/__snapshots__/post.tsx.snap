// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post page matches snapshot 1`] = `
Array [
  <div
    className="sc-AxhCb bqUvYT"
  >
    <header
      className="sc-AxhUy kfXuGD"
    >
      <div
        className="sc-AxheI dMmxDl"
      >
        <a
          className="sc-AxgMl kfMgJa"
          href="/"
        >
          <svg
            className="block"
            fill="none"
            height={50}
            viewBox="0 0 198 198"
            width={50}
          >
            <rect
              fill="var(--color-bg)"
              height={140}
              rx={23}
              transform="rotate(-45 0 98.995)"
              width={140}
              y={98.9949}
            />
            <rect
              height={104.716}
              rx={14.5}
              stroke="var(--color-text)"
              strokeWidth={7}
              transform="rotate(-45 25.452 99.498)"
              width={104.716}
              x={25.4523}
              y={99.4975}
            />
            <path
              d="M78.176 117.776l4.928-.64V79.76l-4.672-.896v-4.416h9.856c.81 0 1.685-.021 2.624-.064a61.142 61.142 0 002.24-.128c2.304-.17 4.267-.256 5.888-.256 4.565 0 8.192.448 10.88 1.344 2.688.853 4.629 2.219 5.824 4.096 1.237 1.835 1.856 4.267 1.856 7.296 0 2.901-.768 5.61-2.304 8.128-1.536 2.517-3.797 4.245-6.784 5.184 1.963.725 3.605 2.091 4.928 4.096 1.365 1.963 2.773 4.48 4.224 7.552.853 1.792 1.536 3.136 2.048 4.032.512.853 1.045 1.344 1.6 1.472l1.856.576V122h-12.16c-.64 0-1.301-.683-1.984-2.048s-1.557-3.499-2.624-6.4c-1.323-3.541-2.475-6.272-3.456-8.192-.939-1.963-1.984-3.157-3.136-3.584h-5.76v15.36l5.696.64V122H78.176v-4.224zm20.8-20.16c2.219 0 3.947-.853 5.184-2.56 1.237-1.75 1.856-4.373 1.856-7.872 0-3.072-.619-5.312-1.856-6.72-1.195-1.45-3.221-2.176-6.08-2.176-1.536 0-2.88.107-4.032.32v18.816c.555.128 1.472.192 2.752.192h2.176z"
              fill="var(--color-text)"
            />
          </svg>
        </a>
        <ul
          className="sc-AxmLO sc-fzozJi gdovsk"
        >
          <li
            className="sc-fzoLsD juCxMN"
          >
            <div
              className="sc-fzpans kLztrW"
            >
              <a
                className="sc-fzplWN laeLlV"
                href="/post"
              >
                Posts
              </a>
            </div>
          </li>
          <li
            className="sc-fzoLsD juCxMN"
          >
            <div
              className="sc-fzpans kLztrW"
            >
              <a
                className="sc-fzplWN laeLlV"
                href="/snippets"
              >
                Snippets
              </a>
            </div>
          </li>
          <li
            className="sc-fzoLsD juCxMN"
          >
            <div
              className="sc-fzpans kLztrW"
            >
              <a
                className="sc-fzplWN laeLlV"
                href="/"
              >
                Newsletter
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="sc-Axmtr fXHeCE"
      >
        <ul
          className="sc-AxmLO eUduLR"
          id="right"
        >
          <li
            className="sc-fzoLsD juCxMN"
          >
            <div
              className="sc-fzpans kLztrW"
            >
              <button
                className="sc-fznyAO gJyGqq"
                onClick={[Function]}
              >
                <svg
                  className="sc-AxjAm fITGcA"
                  height={18}
                  style={
                    Object {
                      "transform": "rotate(40deg)",
                      "transformOrigin": "center center",
                    }
                  }
                  viewBox="0 0 18 18"
                  width={18}
                >
                  <mask
                    id="moon-mask"
                  >
                    <rect
                      fill="#fff"
                      height={18}
                      width={18}
                      x={0}
                      y={0}
                    />
                    <circle
                      cx={10}
                      cy={2}
                      fill="black"
                      r={8}
                    />
                  </mask>
                  <circle
                    cx={9}
                    cy={9}
                    fill="#cfcbc6"
                    mask="url(#moon-mask)"
                    r={8}
                  />
                  <g>
                    <circle
                      cx={17}
                      cy={9}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                    <circle
                      cx={13}
                      cy={15.9282}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                    <circle
                      cx={5}
                      cy={15.9282}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                    <circle
                      cx={1}
                      cy={9}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                    <circle
                      cx={4}
                      cy={2.0717}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                    <circle
                      cx={13}
                      cy={2.0717}
                      fill="#cfcbc6"
                      r="1.5"
                      style={
                        Object {
                          "transform": "scale(0)",
                          "transformOrigin": "center center",
                        }
                      }
                    />
                  </g>
                </svg>
              </button>
            </div>
          </li>
        </ul>
      </div>
    </header>
  </div>,
  <div
    style={
      Object {
        "background": "var(--color-bg)",
        "color": "var(--color-text)",
      }
    }
  >
    <section
      className="sc-fzpjYC kDMHyG"
    >
      <div
        className="sc-fznxsB nKFYp"
      >
        <div>
          <h1>
            1
          </h1>
        </div>
      </div>
    </section>
    <section
      className="sc-fznJRM eubvmk"
    >
      <div
        className="sc-fzoNJl MAkJM"
      >
        <span
          className="sc-fzoXWK fMwCSm"
        >
          <input
            className="sc-fzpmMD llQpjO"
            id="search"
            onChange={[Function]}
            placeholder="Search posts"
            type="search"
            value=""
          />
        </span>
      </div>
      <div
        className="sc-fznWqX cRiCys"
      >
        <h1>
          Newest
        </h1>
        <h4>
          1
          Â  articles
        </h4>
      </div>
      <div
        className="sc-fzoiQi gkFNmM"
      >
        <div
          className="sc-fzqARJ fsrsfw"
        >
          <a
            className="sc-fzoyTs fKcgqV"
            href="/"
          >
            <h4>
              Default Title
            </h4>
            <h6>
              Default description
            </h6>
            <p>
              Some to read
            </p>
            <div
              className="sc-fzqNqU kYepZm"
            >
              <a
                className="sc-fzoLag kOBOPU"
                href="/tags/tags"
              >
                tags
              </a>
            </div>
          </a>
        </div>
      </div>
    </section>
  </div>,
]
`;
